### PR TITLE
[web] sync => isSync , scuba => golden

### PR DESCRIPTION
**Pull request by mdebbar**
This PR addresses some of the items in https://github.com/flutter/flutter/issues/100394

### CODE REVIEW

File: lib/web_ui/test/html/paragraph/text_goldens.dart [hunk 1]
```diff
@@ -13,29 +13,28 @@ import 'package:web_engine_tester/golden_tester.dart';
...
     assert(() {
       if (viewportSize.width.ceil() != viewportSize.width ||
           viewportSize.height.ceil() != viewportSize.height) {
         throw Exception(
-            'Scuba only supports integer screen sizes, but found: $viewportSize');
+            'Gold only supports integer screen sizes, but found: $viewportSize');
       }
       if (viewportSize.width < 472) {
-        throw Exception('Scuba does not support screen width smaller than 472');
+        // TODO: Do we still need this check?
```
Comment (mdebbar) [inline lib/web_ui/test/html/paragraph/text_goldens.dart:31]: @ditman do you know if this is still necessary?

File: lib/web_ui/test/html/paragraph/text_multiline_clipping_golden_test.dart [hunk 2]
```diff
@@ -93,11 +92,8 @@ Future<void> testMain() async {
     (EngineCanvas canvas) {
       // [DomCanvas] doesn't support clip commands.
       if (canvas is! DomCanvas) {
-        // TODO(mdebbar): https://github.com/flutter/flutter/issues/35086
-        // This produces the wrong result when using [BitmapCanvas] but without
-        // the new experimental canvas mode.
```
Comment (mdebbar) [inline lib/web_ui/test/html/paragraph/text_multiline_clipping_golden_test.dart:98]: This was fixed a while ago.

File: lib/web_ui/test/html/paragraph/text_goldens.dart [hunk 3]
```diff
@@ -13,29 +13,28 @@ import 'package:web_engine_tester/golden_tester.dart';
...
+        throw Exception('We do not support screen width smaller than 472');
```
Comment (yjbanov) [inline lib/web_ui/test/html/paragraph/text_goldens.dart:32]: nit: s/We do not/Web engine test harness does not/

**Review by yjbanov — APPROVED** [review]

File: lib/web_ui/test/html/paragraph/text_goldens.dart (hunk 1 above)
Comment (ditman) [reply lib/web_ui/test/html/paragraph/text_goldens.dart:31]: That comment must be from 2019, I don't think we have any limitations of Scuba any longer!

Comment (ditman) [reply lib/web_ui/test/html/paragraph/text_goldens.dart:31]: (I cannot find anything anywhere mentioning 472 pixels anymore, and I don't remember that :P)

File: lib/ui/painting.dart
Comment (ditman) [inline lib/ui/painting.dart:1]: This file is not `[web]` right?

Comment (mdebbar) [reply lib/ui/painting.dart:1]: Good catch! Let me add some of the recent editors of this file.

**Review by dnfield — APPROVED** [review]

File: lib/web_ui/test/html/paragraph/text_goldens.dart (hunk 3 above)
Comment (mdebbar) [reply lib/web_ui/test/html/paragraph/text_goldens.dart:32]: This isn't applicable anymore, I removed this `if` block :)

File: lib/web_ui/test/html/paragraph/text_goldens.dart (hunk 1 above)
Comment (mdebbar) [reply lib/web_ui/test/html/paragraph/text_goldens.dart:31]: Thanks for checking! I'll remove it then.

**Pull request merged**